safe_cli/operators/hw_wallets/trezor_wallet.py

```diff
@@ -51,11 +58,77 @@ def sign_typed_hash(self, domain_hash: bytes, message_hash: bytes) -> bytes:
         )
         return signed.signature
 
-    def get_signed_raw_transaction(self, tx_parameters: TxParams) -> HexStr:
+    @raise_trezor_exception_as_hw_wallet_exception
+    def get_signed_raw_transaction(
+        self, tx_parameters: TxParams, chain_id: int
+    ) -> HexStr:
         """
 
         :param chain_id:
         :param tx_parameters:
         :return: raw transaction signed
         """
-        raise NotImplementedError
+        address_n = tools.parse_path(self.derivation_path)
+        if tx_parameters.get("maxPriorityFeePerGas"):
+            # EIP1559
+            v, r, s = sign_tx_eip1559(
+                self.client,
+                n=address_n,
+                nonce=tx_parameters["nonce"],
+                gas_limit=tx_parameters["gas"],
+                to=tx_parameters["to"],
+                value=tx_parameters["value"],
+                data=HexBytes(tx_parameters["data"]),
+                chain_id=chain_id,
+                max_gas_fee=tx_parameters.get("maxFeePerGas"),
+                max_priority_fee=tx_parameters.get("maxPriorityFeePerGas"),
+            )
+
+            encoded_transaction = (
+                "0x02"
+                + rlp.encode(
+                    [
+                        chain_id,
+                        tx_parameters["nonce"],
+                        tx_parameters.get("maxPriorityFeePerGas"),
+                        tx_parameters.get("maxFeePerGas"),
+                        tx_parameters["gas"],
+                        HexBytes(tx_parameters["to"]),
+                        tx_parameters["value"],
+                        HexBytes(tx_parameters["data"]),
+                        [],
+                        v,
+                        HexBytes(r),
+                        HexBytes(s),
+                    ]
+                ).hex()
+            )
+        else:
+            # Legacy transaction
+            v, r, s = sign_tx(
+                self.client,
+                n=address_n,
+                nonce=tx_parameters["nonce"],
+                gas_price=tx_parameters["gasPrice"],
+                gas_limit=tx_parameters["gas"],
+                to=tx_parameters["to"],
+                value=tx_parameters["value"],
+                data=HexBytes(tx_parameters.get("data")),
+                chain_id=chain_id,
+            )
+
+            encoded_transaction = rlp.encode(
+                [
+                    tx_parameters["nonce"],
+                    tx_parameters["gasPrice"],
+                    tx_parameters["gas"],
+                    HexBytes(tx_parameters["to"]),
+                    tx_parameters["value"],
+                    HexBytes(tx_parameters["data"]),
+                    v,
+                    HexBytes(r),
+                    HexBytes(s),
+                ]
+            ).hex()
+
+        return HexStr(encoded_transaction)
```

Comment:
Shouldn't we return `bytes` or `HexBytes` here? That's what we do everywhere we are returning raw transactions or data, also it's more friendly memory wise

Reply:
Make sense, I was returning `HexStr` to be consistent with ledger response, but I think that is better to be consistent with our code as we are using bytes normally to represent raw_transactions.